deployments/ansible/roles/collector/README.md

```diff
@@ -253,3 +253,89 @@ after installation/configuration in order for any change to take effect.
+  (**default:** `latest`)
+
+- `signalfx_dotnet_auto_instrumentation_msi_url` (Windows only): Specify the
+  download URL to skip the GitHub API request, e.g.
+  `https://github.com/signalfx/signalfx-dotnet-tracing/releases/download/v1.0.0/signalfx-dotnet-tracing-1.0.0-x64.msi`,
+  or to download the MSI from a custom host, e.g.
+  `https://my.host/signalfx-dotnet-tracing-1.0.0-x64.msi`. If specified, the
+  `signalfx_dotnet_auto_instrumentation_version` option is ignored.
+  (**default:** ``)
+
+- `signalfx_dotnet_auto_instrumentation_github_token` (Windows only): Specify
+  a token to authenticate with the GitHub API when making requests to get the
+  latest `signalfx-dotnet-tracing` release. A token is recommended when
+  `signalfx_dotnet_auto_instrumentation_version` is `latest` or when not using
+  `signalfx_dotnet_auto_instrumentation_msi_url` since unauthenticated requests
+  are [rate-limited](https://docs.github.com/en/rest/rate-limit) by GitHub.
+  (**default:** ``)
+
+- `signalfx_dotnet_auto_instrumentation_environment` (Windows only): Configure
+  this option to set the system-wide "Environment" value to be reported to
+  Splunk APM, e.g. `prod`. The value is assigned to the `SIGNALFX_ENV`
+  environment variable in the Windows registry. (**default:** ``, i.e. the
+  "Environment" will appear as `unknown` in Splunk APM for the instrumented
+  service/application)
+
+- `signalfx_dotnet_auto_instrumentation_service_name` (Windows only): Configure
+  this variable to override the [auto-generated service name](
+  https://docs.splunk.com/Observability/gdi/get-data-in/application/dotnet/configuration/advanced-dotnet-configuration.html#changing-the-default-service-name)
+  for the instrumented service/application, e.g. `my-service-name`. The value
+  is assigned to the `SIGNALFX_SERVICE_NAME` environment variable in the
+  Windows registry. (**default:** ``)
+
+- `signalfx_dotnet_auto_instrumentation_enable_profiler` (Windows only): Set
+  this option to `true` to enable AlwaysOn CPU Profiling. The value will be
+  assigned to the `SIGNALFX_PROFILER_ENABLED` environment variable in the
+  Windows registry. (**default:** `false`)
+
+- `signalfx_dotnet_auto_instrumentation_enable_profiler_memory` (Windows only):
+  Set this option to `true` to enable AlwaysOn Memory Profiling. The value will
+  be assigned to the `SIGNALFX_PROFILER_MEMORY_ENABLED` environment variable in
+  the Windows registry. (**default:** `false`)
+
+- `signalfx_dotnet_auto_instrumentation_iisreset` (Windows only): By default,
+  the `iisreset.exe` Powershell command will be executed after
+  installation/configuration in order for auto instrumentation to take effect
+  for IIS applications. Set this option to `false` to skip this step if IIS is
+  managed separately or not applicable. (**default:** `true`)
+
+- `signalfx_dotnet_auto_instrumentation_additional_options` (Windows only):
```

Comment:
@pjanotti Is this generic option sufficient for adding the `COR_*` and `CORECLR_*` env vars to the registry?  Or should we add more specific options to set those, similar to the `signalfx_dotnet_auto_instrumentation_enable_profiler` option for configuring `SIGNALFX_PROFILER_ENABLED`?

Reply:
Because these 4 env vars have a system-wide impact (causing all .NET apps to be instrumented) I think it is better to have something like `signalfx_dotnet_auto_instrumentation_system_wide` that calls attention that it will instrument all .NET apps on the box. It should mention that the user should consider using `signalfx_dotnet_auto_instrumentation_additional_options` to disable the instrumentation of some apps that they may don't want to be instrumented. That is done by using the
`SIGNALFX_PROFILER_EXCLUDE_PROCESSES` and `SIGNALFX_PROFILER_PROCESSES`
environment variables to include/exclude applications from the tracing auto-instrumentation.
These are ";" delimited lists that control the inclusion/exclusion of processes.

Reply:
@pjanotti I added the `signalfx_dotnet_auto_instrumentation_system_wide` option to add the env vars in `HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\Session Manager\Environment` (please see the updated readme for details).

Reply:
@jeffreyc-splunk the options are looking good. I think the text describing them could be improved, but, I will leave that part of the review to our (excellent) docs team :)